lib/classroom_api.rb

```diff
@@ -222,79 +261,92 @@ def create_classroom_characteristics(characteristics)
+      number_of_api_calls_per_minutes = 0
+      redo_loop_number = 1
+      classrooms.each do |room|
+        if number_of_api_calls_per_minutes < 400
+          number_of_api_calls_per_minutes += 1
+        else
+          @log.api_logger.debug "update_all_classroom_contacts, the script sleeps after #{number_of_api_calls_per_minutes} calls"
+          number_of_api_calls_per_minutes = 1
+          sleep(61.seconds)
         end
-        redo_loop_number += 1
-        redo
-      elsif result['success']
-        if result['data']['Classrooms'].present?
-          row = result['data']['Classrooms']['Classroom'][0]
-          
-          if RoomContact.find_by(rmrecnbr: rmrecnbr).present?
-            update_classroom_contact(row, rmrecnbr)
+        facility_id = room.facility_code_heprod
+        rmrecnbr = room.rmrecnbr
+        result = get_classroom_contact(ERB::Util.url_encode(facility_id))
+        if result['errorcode'] == "ERR429"
+          @log.api_logger.debug "update_all_classroom_contacts, error: API return: #{result['errorcode']} - #{result['error']} after #{number_of_api_calls_per_minutes} calls"
+          number_of_api_calls_per_minutes = 0
+          sleep(61.seconds)
+          if redo_loop_number > 9
+            @debug = true
+            @log.api_logger.debug "update_all_classroom_contacts, error: API return: #{result['errorcode']} - #{result['error']} after #{number_of_api_calls_per_minutes} calls #{redo_loop_number} times "
+            return @debug
+          end
+          redo_loop_number += 1
+          redo
+        elsif result['success']
+          if result['data']['Classroom'].present?
+            row = result['data']['Classroom'][0]
+            
+            if RoomContact.find_by(rmrecnbr: rmrecnbr).present?
+              update_classroom_contact(row, rmrecnbr)
+            else
+              create_classroom_contact(row, rmrecnbr)
+            end
           else
-            create_classroom_contact(row, rmrecnbr)
+            @log.api_logger.info "update_all_classroom_contacts, No contacts for facility_id #{facility_id}"
           end
         else
-          @log.api_logger.info "update_all_classroom_contacts, No contacts for facility_id #{facility_id}"
+          @log.api_logger.debug "update_all_classroom_contacts, error: API returns false for facility_id #{facility_id}: #{result['errorcode']} - #{result['error']}"
+          @debug = true
+          return @debug
         end
-      else
-        @log.api_logger.debug "update_all_classroom_contacts, error: API returns false for facility_id #{facility_id}: #{result['errorcode']} - #{result['error']}"
-        @debug = true
-        return @debug
+        return @debug if @debug
       end
-      return @debug if @debug
+    rescue StandardError => e
+      # example: Errno::ETIMEDOUT: Operation timed out - user specified timeout
+      @log.api_logger.debug "update_rooms, error: API return: #{e.message}"
```

Comment:
Consider updating the message to say _**update_all_classroom_contacts**_ instead of _**update_rooms**_

Reply:
I don't think it's necessary, but I did it here
done in here [e23935c](https://github.com/lsa-mis/mi_classrooms/pull/326/commits/e23935c49e6db3b2e0b0e8d728d91a3e359e659e)